Minor silicon improvements

Ms-Mee opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
This PR does the following:
Makes it possible for silicons to gain mining skill. Adds skill upgrades for borgs that mas the skill of their focus, comes alongside the respective research or illegal tech for gaming skill.
Makes AIs able to speak in shortwave mode and makes borgs capable of expanding ships.
Demodularizes the shuttle area editor file.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Borgs tend to get outclassed due to skill progression and a few other things they lack the ability to perform that are minor, such as stopping gibtonite explosions. They usually should excel at their function, but get left behind even though they are specialized, due to it being impossible for mining cyborgs to gain mining skill and medical cyborgs not needing to do surgery too much and as such not gaining a whole lot of medical skill. ~~Gaming skill upgrade funny.~~
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: New cyborg upgrades that max their skill on whichever skill reflects their module, if any does.
fix: Silicons are now properly capable of receiving mining skill.
balance: The cyborg blueprints are now expansion permits, which should allow them to expand ships.
balance: AIs are now capable of communicating with their built-in radios, which is now capable of swappping out of subspace mode. They no longer default to subspace mode when deployed to their shells.
code: Demodularizes shuttle area editors.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Ms-Mee committed:
Fun stuff

Ms-Mee committed:
Update robot_upgrades.dm

Ms-Mee committed:
DN

Ms-Mee committed:
Yeah.

Ms-Mee committed:
Bruh momentum

Ms-Mee committed:
...

Ms-Mee committed:
Merge branch 'shiptest-ss13:master' into BorgAdjust

Ms-Mee committed:
Merge branch 'shiptest-ss13:master' into BorgAdjust

Ms-Mee committed:
Now they can communicate

Ms-Mee committed:
Merge branch 'BorgAdjust' of https://github.com/Ms-Mee/Shiptest into BorgAdjust

Ms-Mee committed:
Makes them not default to the unusable mode

Ms-Mee committed:
Better idea actually

Ms-Mee commented:
True github actions bot, I should demodularize the area editors

Ms-Mee committed:
Demodularization is sus!

Ms-Mee committed:
Delete blueprints.dm

Ms-Mee committed:
Merge branch 'BorgAdjust' of https://github.com/Ms-Mee/Shiptest into BorgAdjust

Ms-Mee reviewed (commented):
And done, that.

Ms-Mee committed:
Bye bye

Kapu1178 commented on code/game/objects/items.dm:
```diff
@@ -875,8 +875,11 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
+		var/mob/living/silicon/H = user
+		skill_modifier = H.mind.get_skill_modifier(/datum/skill/mining, SKILL_SPEED_MODIFIER)
 
+	delay *= toolspeed * skill_modifier
```
this is snowflakey and bad

Ms-Mee replied:
If I'm not mistaken it works fine without either checks but it's been a long time since I tested so I'm not too inclined to trust remembering that properly, if it does work I have to wonder why the check is even there in the first place

Ms-Mee replied:
Upon further investigation it's even more odd to me as the special actions some could argue shouldn't give exp such as simplemobs smashing rocks... They already don't give exp because give_exp defaults to false, this is literally here to prevent borgs from getting mining exp it seems, unless I missed something

Ms-Mee replied:
Or rather this one is to prevent any non-human mobs from getting the bonuses, the exp is given elsewhere

Kapu1178 commented on code/game/objects/items/blueprints.dm:
```diff
@@ -208,6 +208,202 @@
+		var/obj/machinery/computer/helm/H = target
+		if(istype(H.current_ship, /obj/structure/overmap/ship/simulated))
+			var/obj/structure/overmap/ship/simulated/S = H.current_ship
+			target_shuttle = S.shuttle
```
```suggestion
	if(istype(target, /obj/machinery/computer/helm) && istype(H.current_ship, /obj/structure/overmap/ship/simulated))
		var/obj/machinery/computer/helm/H = target
		var/obj/structure/overmap/ship/simulated/S = H.current_ship
		target_shuttle = S.shuttle
```

Kapu1178 commented on code/game/objects/items/blueprints.dm:
```diff
@@ -208,6 +208,202 @@
+		if(creator.create_area_cooldown >= world.time)
+			to_chat(creator, "<span class='warning'>You're trying to create a new area a little too fast.</span>")
+			return
+		creator.create_area_cooldown = world.time + 10
```
```suggestion
	if(!creator)
		return

	if(creator.create_area_cooldown >= world.time)
		to_chat(creator, "<span class='warning'>You're trying to create a new area a little too fast.</span>")
		return
	creator.create_area_cooldown = world.time + 10
```

Kapu1178 commented on code/game/objects/items/blueprints.dm:
```diff
@@ -208,6 +208,202 @@
+
+	var/list/firedoors = oldA.firedoors
+	for(var/door in firedoors)
+		var/obj/machinery/door/firedoor/FD = door
```
```suggestion
	for(var/obj/machinery/door/firedoor/FD as anything in firedoors)
```

Kapu1178 commented on code/game/objects/items/blueprints.dm:
```diff
@@ -208,6 +208,202 @@
+	target_shuttle.shuttle_areas[newA] = TRUE
+
+	newA.connect_to_shuttle(target_shuttle, target_shuttle.get_docked())
+	for(var/atom/thing in newA)
```
```suggestion
	for(var/atom/thing as anything in newA)
```

As far as im aware areas can only contain ATOMs unless theres some really cursed shit I don't know about.

Kapu1178 commented:
Marked DNM until Mark is well enough to review shuttle code.

Ms-Mee commented:
> Marked DNM until Mark is well enough to review shuttle code.

It *is* what was already there I just moved it out of the modular folder and added a borg variant, as such any changes are outside of my knowledge as I'm not familiarized enough with the code to have any confidence in changing it, nor is changing it within the initial scope of this PR but if it is decided such would be best sure

Kapu1178 reviewed (changes requested):
i didnt mean to approve

Ms-Mee committed:
Bang boom

Ms-Mee committed:
E

Ms-Mee committed:
Merge branch 'shiptest-ss13:master' into BorgAdjust

Ms-Mee commented:
Well I'm waiting too, github action bot, I want to get my gamer skill upgrade someday too.

Ms-Mee committed:
Merge branch 'shiptest-ss13:master' into BorgAdjust

triplezeta commented:
@MarkSuckerberg

ZephyrTFA committed:
apply code adjustments as pr author is asleep

Co-authored-by: Kapu1178 <75460809+Kapu1178@users.noreply.github.com>

MarkSuckerberg committed:
fixes error

Ms-Mee commented:
You gotta be kidding me I just turned off my computer

Zevotech commented:
oh yeah unironically merge this im begging you

Ms-Mee commented:
> oh yeah unironically merge this im begging you

Gotta fix the merge conflict and about the demodularization part, when I have the time to

Ms-Mee committed:
Update

Ms-Mee committed:
Just this I guess.

Ms-Mee committed:
Merge branch 'master' into BorgAdjust

Ms-Mee committed:
Solve duplicate

Ms-Mee committed:
Minot thing

Ms-Mee commented:
Later today I shall review and retest this PR to see the state it is at proper and finish it for once.

Ms-Mee commented:
I forgor 💀
